### ISSUE TITLE
Question with 'Replace missing values with' = 'Nothing' is broken after upgrade to 0.29.3

### ISSUE THREAD

**Issue by ivashin** [metabase/metabase#7773]
- Your browser and the version: Firefox 60.0.1, Chrome 67
- Your operating system: Ubuntu 16.04
- Your databases: MySQL, Google Analytics
- Metabase version: 0.29.3
- Metabase hosting environment: Docker on AWS ECS
- Metabase internal database: MySQL

We have a query built in query builder with a line plot that has 'Replace missing values with' = Nothing setting. It worked fine in 0.28.6, but after upgrading to 0.29.3 this plot gets stuck in JS code and becomes unresponsive. This makes dashboard with this question unresponsive as well.

I've created the same question from scratch in 0.29.3 and it works fine until you go into plot settings and change 'Replace missing values with' to Nothing - then the page immediately becomes unresponsive.

Generated SQL for the question is:
```
SELECT max(`failure_metrics`.`max_failed_percentage`) AS `max`, sum(`failure_metrics`.`failed_count`) AS `sum`, str_to_date(date_format(`failure_metrics`.`timestamp`, '%Y-%m-%d %H:%i'), '%Y-%m-%d %H:%i') AS `timestamp`
FROM `failure_metrics`
WHERE str_to_date(date_format(`failure_metrics`.`timestamp`, '%Y-%m-%d %H'), '%Y-%m-%d %H') BETWEEN str_to_date(date_format(date_add(now(), INTERVAL -24 hour), '%Y-%m-%d %H'), '%Y-%m-%d %H')
   AND str_to_date(date_format(now(), '%Y-%m-%d %H'), '%Y-%m-%d %H')
GROUP BY str_to_date(date_format(`failure_metrics`.`timestamp`, '%Y-%m-%d %H:%i'), '%Y-%m-%d %H:%i')
ORDER BY str_to_date(date_format(`failure_metrics`.`timestamp`, '%Y-%m-%d %H:%i'), '%Y-%m-%d %H:%i') ASC
```

Performance analysis snapshot is below:
```
19,211.45 ms | 94.08% | 19,211.45 ms | 94.08% | 16880 | createIndexFinder/<vendor.bundle.js:1047
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |  _.includevendor.bundle.js:667
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |   _.uniquevendor.bundle.js:924
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |    _.mixin/</_.prototype[name]vendor.bundle.js:1895
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |     getXValuesapp-main.bundle.js:5319
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |      notifyAllvendor.bundle.js:84833
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |       closevendor.bundle.js:127996
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |        closeAllvendor.bundle.js:63054
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |         performvendor.bundle.js:62989
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |          performvendor.bundle.js:51303
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |           flushBatchedUpdatesvendor.bundle.js:51381
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |            closeAllvendor.bundle.js:63054
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |             performvendor.bundle.js:62989
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |              batchedUpdatesvendor.bundle.js:127662
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |               batchedUpdatesvendor.bundle.js:51312
19,211.45 ms | 94.08% | 0 ms | 0% | 0 |                dispatchEventvendor.bundle.js:127818
```

**Comment by ivashin** [comment]
Found one more possibly related issue: question with 'Replace missing values with' = 'Zero' shows zeroes everywhere instead of actual values.

**Comment by sabino** [comment]
We are having some issue with this as well, but the page doesn't freeze.

When using `Timeseries` combined with **Replace missing values with** `Nothing` the chat doesn't show up, when using `Zero` all numbers get replaced by zeros.

Using a Postgresql connection.
Steps to reproduce:

```sql
SELECT '2018-01-01' as month, 2 as num
UNION ALL
SELECT '2018-02-01' as month, 3 as num
UNION ALL
SELECT '2018-04-01' as month, null as num
UNION ALL
SELECT '2018-07-01' as month, 2 as num
```

1. Set X-axis to Timestamp
2. Set Replace missing values with `Nothing` or `Zero`


- When using `Linear Interpolated`:
<img width="1022" alt="when-using-linear-interpolated" src="https://user-images.githubusercontent.com/982190/41251021-078cc43e-6d8f-11e8-9213-cf068a736fd0.png">

- When using `Nothing`
<img width="1043" alt="nothing" src="https://user-images.githubusercontent.com/982190/41251053-2add0232-6d8f-11e8-9ac0-9f9834143012.png">

- When using `Zero`
<img width="1062" alt="zeros" src="https://user-images.githubusercontent.com/982190/41251071-3d211adc-6d8f-11e8-8c84-d35f0537ac47.png">

**Comment by doodlebro** [comment]
Why does this keep getting broken off and on?

**Comment by ccbrandenburg** [comment]
facing the same issue on 0.30.4

**Comment by lp-lima** [comment]
Similar issue here... 
There is data missing between 2018-11-12 and 2018-11-20:
![image](https://user-images.githubusercontent.com/22321989/49962307-13129a00-fefc-11e8-8955-3f819d94d27b.png)

When I choose to replace missing values with zero:
![image](https://user-images.githubusercontent.com/22321989/49962451-6b499c00-fefc-11e8-943a-b151f2413411.png)



I don't know if this is the place to post this, but, also, the same question, when written in SQL, returns dates not in the correct format, despite the return datatype from MySQL being explicitly date (by using a date() function)... therefore this

![image](https://user-images.githubusercontent.com/22321989/49962625-d3987d80-fefc-11e8-9278-6a15863a0808.png)
![image](https://user-images.githubusercontent.com/22321989/49962713-23774480-fefd-11e8-8e9e-31f87446c9cf.png)


becomes this in a line chart:

![image](https://user-images.githubusercontent.com/22321989/49962743-3a1d9b80-fefd-11e8-8d2a-b553075b116d.png)

Again, there is missing data. When I try to replace it with zero:

![image](https://user-images.githubusercontent.com/22321989/49962778-50c3f280-fefd-11e8-81f2-24de6189010c.png)

What is happening here? Well, it is replacing missing values for zero correctly - the real problem being, though, that it is considering my X axis as being datetime, not date, even though I clearly passed it a date field. This might be a different issue so - if so, lemme know.

**Comment by flamber** [comment]
Isn't this a dupe of #4802 ?

**Comment by flamber** [comment]
I cannot reproduce unresponsiveness on 0.35.3. I think it was fixed in 0.33 (if not earlier).
